modules/Microsoft.DigitalTwins/digitalTwinsInstances/.test/common/dependencies.bicep

```diff
@@ -0,0 +1,166 @@
+    properties: {
+        addressSpace: {
+            addressPrefixes: [
+                addressPrefix
+            ]
+        }
+        subnets: [
+            {
+                name: 'defaultSubnet'
+                properties: {
+                    addressPrefix: addressPrefix
+                    serviceEndpoints: [
+                        {
+                            service: 'Microsoft.KeyVault'
+                        }
+                    ]
+                }
+            }
+        ]
+    }
+}
+
+resource managedIdentity 'Microsoft.ManagedIdentity/userAssignedIdentities@2018-11-30' = {
+    name: managedIdentityName
+    location: location
+}
+
+resource privateDNSZone 'Microsoft.Network/privateDnsZones@2020-06-01' = {
+    name: 'privatelink.vaultcore.azure.net'
+    location: 'global'
+
+    resource virtualNetworkLinks 'virtualNetworkLinks@2020-06-01' = {
+        name: '${virtualNetwork.name}-vnetlink'
+        location: 'global'
+        properties: {
+            virtualNetwork: {
+                id: virtualNetwork.id
+            }
+            registrationEnabled: false
+        }
+    }
+}
+
+resource eventHubNamespace 'Microsoft.EventHub/namespaces@2022-10-01-preview' = {
+    name: eventHubNamespaceName
+    location: location
+    properties: {
+        zoneRedundant: false
+        isAutoInflateEnabled: false
+        maximumThroughputUnits: 0
+    }
+}
+
+resource eventHub 'Microsoft.EventHub/namespaces/eventhubs@2022-10-01-preview' = {
+    name: eventHubName
+    parent: eventHubNamespace
+}
+
+resource serviceBus 'Microsoft.ServiceBus/namespaces@2022-10-01-preview' = {
+    name: serviceBusName
+    location: location
+    properties: {
+        zoneRedundant: false
+    }
+}
+
+resource serviceBusTopic 'Microsoft.ServiceBus/namespaces/topics@2022-10-01-preview' = {
+    name: serviceBusTopicName
+    parent: serviceBus
+}
+
+resource eventGridDomain 'Microsoft.EventGrid/domains@2022-06-15' = {
+    name: eventGridDomainName
+    location: location
+    properties: {
+        disableLocalAuth: false
+    }
+}
+
+resource eventGridTopic 'Microsoft.EventGrid/domains/topics@2022-06-15' = {
+    name: eventGridTopicName
+    parent: eventGridDomain
+}
+
+resource evhrbacAssignment 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
+    name: guid(managedIdentity.id, 'evhrbacAssignment')
+    scope: eventHubNamespace
+    properties: {
+        roleDefinitionId: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '2b629674-e913-4c01-ae53-ef4638d8f975')
+        principalId: managedIdentity.properties.principalId
+        principalType: 'ServicePrincipal'
+    }
+}
+
+resource sbhrbacAssignment 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
+    name: guid(managedIdentity.id, 'sbrbacAssignment')
+    scope: serviceBus
+    properties: {
+        roleDefinitionId: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '69a216fc-b8fb-44d8-bc22-1f3c2cd27a39')
```

Comment:
Please add a comment in the back what this role is

Reply:
Add comment to explain role id